### PR TITLE
Put the full path to the config.h file.

### PR DESCRIPTION
**Pull request by clalancette**
CMake 3.20 complains if you give it files it can't find.
We only see this on Windows right now since that is the
only platform with CMake 3.20.

It must have been the case that this file couldn't be found
all along, but we just never noticed it.  Fix it here by
giving the full path to where the file is generated.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

This should fix the last CMake warning on the nightly Windows builds.

### CODE REVIEW

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14278/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9067/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11962/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14378/)

File: orocos_kdl/src/CMakeLists.txt [hunk 1]
```diff
@@ -101,7 +101,7 @@ IF(NOT (CMAKE_VERSION VERSION_LESS 2.8.12))
 ENDIF()
 #####end RPATH
 
-ADD_LIBRARY(orocos-kdl ${LIB_TYPE} ${KDL_SRCS} config.h)
+ADD_LIBRARY(orocos-kdl ${LIB_TYPE} ${KDL_SRCS} "${CMAKE_CURRENT_BINARY_DIR}/config.h")
```
Comment (ivanpauno) [inline orocos_kdl/src/CMakeLists.txt:104]: is `config.h` needed here at all?

[This](https://github.com/ros2/orocos_kinematics_dynamics/blob/8516c84bedb8afda4332295550ead5752d9d674d/orocos_kdl/src/CMakeLists.txt#L133-L134) and [this](https://github.com/ros2/orocos_kinematics_dynamics/blob/8516c84bedb8afda4332295550ead5752d9d674d/orocos_kdl/src/CMakeLists.txt#L113) line are enough IMO (actually that `set_target_properties()` line is weird, but it doesn't matter for this PR).

**Review by ivanpauno — APPROVED** [review]
LGTM with green CI, I also think that `config.h` could be removed from that list.

File: orocos_kdl/src/CMakeLists.txt (hunk 1 above)
Comment (clalancette) [reply orocos_kdl/src/CMakeLists.txt:104]: I think the reason this is here is to ensure that if `config.h.in` was changed, this library would be rebuilt.  But I'm not sure about that.

I think I'm going to go ahead and merge this as-is, as it is the minimal patch to fix this.  We can always come back later on and remove it.

Comment (hidmic) [reply orocos_kdl/src/CMakeLists.txt:104]: I think @ivanpauno's right. Compilers can typically generate files to track header dependencies. I'd be surprised if CMake generators do not do that. The fact it is necessary may suggest there's a different problem lurking.

Having said, this change should be harmless

**Review by hidmic — COMMENTED** [review]
LGTM

**Pull request merged**